pwndbg/commands/search.py

```diff
@@ -126,8 +126,19 @@ def search(type, hex, string, executable, writable, value, mapping, save, next):
 
     # Null-terminate strings
     elif type == 'string':
+        value = value.encode()
         value += b'\x00'
 
+    # Find the mappings that we're looking for
+    mappings = pwndbg.vmmap.get()
+
+    if mapping:
+        mappings = [m for m in mappings if mapping in m.objfile]
```

Comment:
`mapping` is a string, e.g. `"libc.so"`.

`mappings` is a list of `pwndbg.memory.Page` objects which describe memory mappings and their regions.